Clinical-Genomics/scout
# Auto-replace PanelApp panels

Fix #3399

Given [this scenario](https://github.com/Clinical-Genomics/scout/issues/3327#issuecomment-1153336049): 
- genes missing in genes database collection
- PanelApp panels runned immediately after updating genes which results in wrong gene panels saved to database
- If, after fixing the genes collection, one runs the same command again, then there is an error triggered by a check for panels aready existing with the same version and **PanelApp panels can't be fixed**.

This PR allows PanelApp panels to be replaced if already existing whenever one runs again the command above

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>


**How to test, locally**:
1. Given a demo scout app running locally.
2. On master branch run:` scout --demo load panel --panel-app --panel-id 522`
3. Run it again and you should get error
4. Switch to this branch and repeat command
5. Panel should be overwritten

**Expected outcome**:
Gene panel documents for PanelApp panels should be replaced in database

**Review:**
